nixos/modules/services/networking/dnsmasq.nix

```diff
@@ -81,6 +110,14 @@ in
 
   config = mkIf cfg.enable {
 
+    warnings = lib.optional (cfg.extraConfig != "") "Dnsmasq now supports `services.dnsmasq.config` for an attribute-set based config";
```

Comment:
we should be more specific maybe and say this is going to be deprecated?

Reply:
I've made a couple of changes, including this, which I've also now mentioned in the release note (if we are going to deprecate something  thought it should be mentioned).